test/distributed/org/apache/cassandra/distributed/test/IndexAndMVDroppingTest.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.distributed.test;
+
+import org.junit.Test;
+
+import org.apache.cassandra.distributed.Cluster;
+
+import static org.apache.cassandra.distributed.Cluster.build;
+import static org.apache.cassandra.distributed.api.ConsistencyLevel.ONE;
+import static org.apache.cassandra.distributed.api.Feature.NETWORK;
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Tests CASSANDRA-18105.
+ * <p>
+ * Trucation will insert an entry to system.local into truncated_at map for given table. Because id of an index
+ * is same as id of the base table, when dropping an index, it in fact drops a table with same id as the base table
+ * which will remove an entry for trucated_at column. Hence, after restart of the node, querying the base table
+ * will resurrect the data because commitlog was fully replayed, not taking into consideration (now removed) entry
+ * in truncated_at column.
+ * <p>
+ * The fix consists of not removing an entry in trucated_at column in system.local
+ * if the table being removed is an index.
+ */
+public class IndexAndMVDroppingTest extends TestBaseImpl
+{
```

Comment:
nit: We could cut down running time a little if we share the cluster. Something like...

```
private static Cluster CLUSTER;

@BeforeClass
public static void init() throws IOException
{
    CLUSTER = init(Cluster.create(1, conf -> conf.with(NETWORK)));
}

@AfterClass
public static void shutdown()
{
    if (CLUSTER != null)
        CLUSTER.close();
}

@After
public void reset()
{
    if (CLUSTER.get(1).isShutdown())
        CLUSTER.get(1).startup(); // make sure we get the cluster running for test that follow a poorly timed failure
}
```